services/src/main/java/io/meeds/gamification/service/DefaultProgramRegistry.java

```diff
@@ -0,0 +1,307 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.meeds.common.ContainerTransactional;
+import io.meeds.gamification.constant.EntityStatusType;
+import io.meeds.gamification.constant.EntityType;
+import io.meeds.gamification.model.*;
+import io.meeds.gamification.model.filter.ProgramFilter;
+import io.meeds.gamification.utils.Utils;
+import io.meeds.social.translation.service.TranslationService;
+import jakarta.annotation.PostConstruct;
+import lombok.SneakyThrows;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.exoplatform.commons.exception.ObjectNotFoundException;
+import org.exoplatform.commons.file.model.FileItem;
+import org.exoplatform.commons.file.services.FileService;
+import org.exoplatform.container.PortalContainer;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.resources.LocaleConfigService;
+import org.exoplatform.services.resources.ResourceBundleService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+public class DefaultProgramRegistry {
+
+  private static final Log                  LOG                  = ExoLogger.getLogger(DefaultProgramRegistry.class);
+
+  @Autowired
+  private ProgramService                    programService;
+
+  @Autowired
+  private RuleService                       ruleService;
+
+  @Autowired
+  private TranslationService                translationService;
+
+  @Autowired
+  private LocaleConfigService               localeConfigService;
+
+  @Autowired
+  private ResourceBundleService             resourceBundleService;
+
+  @Autowired
+  private FileService                       fileService;
+
+  private final List<DefaultRuleConfig>     ruleAvecPrerequisite = new ArrayList<>();
+
+  private final Map<Locale, ResourceBundle> bundles              = new ConcurrentHashMap<>();
+
+  @PostConstruct
+  public void init() {
+    start();
+  }
+
+  @ContainerTransactional
+  public void start() {
+    ProgramFilter filter = new ProgramFilter();
+    filter.setIncludeDeleted(true);
+    filter.setStatus(EntityStatusType.ALL);
+    int programsCount = programService.countPrograms(filter);
+    if (programsCount == 0) {
+      loadDefaultPrograms();
+    }
+  }
+
+  @SneakyThrows
+  private void loadDefaultPrograms() {
+    ObjectMapper mapper = new ObjectMapper();
+    InputStream programInputStream = new ClassPathResource("default-programs.json").getInputStream();
+    List<DefaultProgramConfig> programConfigs = mapper.readValue(programInputStream, new TypeReference<>() {
```

Comment:
Some questions to help you improve the import algorithm:

1. How to determine whether the program has been already imported before or not ?
2. How to determine for each action which is the associated program independently from its I18N label ?

I invite you to give a look at how the data injection is made in service [PortletInstanceImportService](https://github.com/Meeds-io/layout/blob/develop/layout-service/src/main/java/io/meeds/layout/service/injection/PortletInstanceImportService.java) which will import (as here **programs & actions**) **categories && instances** from two different type of files `portlet-instance-categories.json` and `portlet-instances.json`. In few words, an id/name used for import process only is associated to the category/instance which will be the unique identifier that will be used to store/read flag from `SettingService` to know whether the data item has been imported or not. This same id/name will be used to reference the associated category to a given instance to import.

Let me know if you need further information.